_posts/2019-12-18-road-to-scala-3.md

```diff
@@ -0,0 +1,174 @@
+
+### I'm reminded of Python 2 / 3. Should I be worried?
+
+In a word, "no". We believe that Scala's static type system and the compatibility decisions and guarantees described in this document will allow the transition to Scala 3 to be both quicker and less painful than Python's was.
+
+However, we remain vigilant and appreciate any feedback on how we can improve our upgrade strategy. 
+
+Our strategy for making your upgrade predictable and manageable comprises:
+
+1. There will be no "silent" breaking changes in run-time behavior. The standard library is shared. The ABI of Scala 2.13 and 3.0 will be the same: the encoding of key language constructs, such as traits and lazy vals, are shared.
+
+2. Scala 3 is backwards compatible with Scala 2, except for a few deprecated features that are being dropped. The Scala standard library already cross-compiles between the two versions. So can your project.
+
+3. As always, the type checker has your back: changes are often caught by the compiler as type errors or migration warnings.
+
+### How can I contribute?
+
+Perhaps you've been waiting for the design of Scala 3 to settle down before diving in.  That settling-down is now largely complete, so now is an excellent time to get involved. If you're a Dotty first-timer, read the [getting started page](https://dotty.epfl.ch/docs/usage/getting-started.html).
+
+There are two basic ways you can help Scala 3 happen:
+
+1. Help the ecosystem migrate.
+
+2. Participate in the development of Scala itself.
+
+Details on both options follow.
+
+#### Migrating the ecosystem
+
+The biggest thing Scala 3 needs from the community is for everyone to begin porting their code. For established projects, the logical first step would be to start cross-building against the latest  Scala 3 milestone, when they become available (or use a Dotty milestone right now!). We expect this to be a common scenario during the transition period. Or, you can jump straight in and upgrade your code to Scala 3 and try out some of the new features!
+
+We're especially interested in knowing if something blocks you completely, but other other kinds of feedback are welcome as well.
+
+Every Scala user can participate, but it's especially helpful for maintainers of open-source libraries and tooling to begin moving.  If you are a maintainer, then as an initial step, please consider adding Dotty to your project's cross-build
+
+In the simplest cases, you just need to:
+
+* add sbt-dotty to `project/plugins.sbt`
+
+* add the latest Dotty version to `crossScalaVersions` and `.travis.yml`
+
+* perhaps adjust your compiler flags, according to `isDotty.value`, e.g. to pass `-language:Scala2` to Dotty
```

Comment:
There was thought of renaming that flag into `Scala2Compat` instead of just `Scala2` not to give the wrong expectations about what it does. Was this done?

Reply:
Let's take this discussion outside of the post.